src/main/java/com/criteo/nosql/mewpoke/memcached/MemcachedMonitor.java

```diff
@@ -73,9 +74,49 @@ private MemcachedMonitor(final String serviceName, final MemcachedClient client,
         return (Map) client.getStats();
     }
 
+
+
     public Map<InetSocketAddress, Map<String, String>> collectStatsItems() {
         // TODO: re-implement get stats in order to avoid creating new map each time
-        return (Map) client.getStats("items");
+        final Map<SocketAddress, Map<String, String>> itemsStats = client.getStats("items");
+        final Map<SocketAddress, Map<String, String>> slabsStats = client.getStats("slabs");
+        final Map<SocketAddress, Map<String, String>> nodeSettings = client.getStats("settings");
+
+        // We return slab size range instead of slabid
+        // Need to re-construct client.getStats
+        final Map<SocketAddress, Map<String, String>> itemsStatsByRange = new HashMap<>();
+
+        itemsStats.forEach((socketAddr, itemStats) -> {
+
+            // Construct Map of slabid:slab range size for each node
+            final Map<String, Double> minSizeBySlabId = new HashMap<>();
+            slabsStats.get(socketAddr).forEach((statName, statValue) -> {
+                if(statName.matches("(.*):chunk_size")){
```

Comment:
In Java String.split() takes a regex in parameter which can be slow (regex need to be parsed/compiled) for simple char separator.
You can use guava (google java library) to split by simple char https://google.github.io/guava/releases/23.0/api/docs/com/google/common/base/Splitter.html
If we don't have already guava as a dependency, forget about my comment